NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -45,4 +46,82 @@ class MainPageViewController: UIViewController {
     @IBAction func skipButtonTapped(_ sender: Any) {
         updateProfileIndex()
     }
+    
+    func getUserData(){
+        let request = createGetProfileRequest()
+        processSessionTasks(request: request!, callback: readReponse)
+    }
+    
+    func parseUserData(json: NSDictionary){
+
+    }
+    
+    func readReponse(parseJSON: NSDictionary) -> Void{
+        print(parseJSON)
+        let msg = parseJSON["success"] as? Bool
+        if msg == false {
+            print("Registration Data Failed to Send")
+            return
+        }
+    }
+    
+    func encodeParams(raw_params: Dictionary<String, String>) -> String{
+        var params:String = ""
+        var n = 0
+        for (key, value) in raw_params{
+            if n == 0{
+                params = "?" + key + "=" + value
+            }else{
+                params = params + "&" + key + "=" + value
+            }
+            n += 1
+        }
+        return params
+    }
+    
+    func createGetProfileRequest() -> URLRequest? {
+        // TODO: get username and api_key from keychain/local storage
+        let psudo_params = [
+            "username":"duckmoll",
+            "api_key":"f1c085c2ed2196cb029827c80f879a2ef3ce2189"
+        ]
+        let params = encodeParams(raw_params: psudo_params)
+        let url = getURL(path: "profile/" + params)
+        var request = URLRequest(url:url)
+        request.httpMethod = "GET"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("ApiKey", forHTTPHeaderField: "Authorization")
+        return request
+    }
+    
+    struct User:Decodable{
+        let date_joined: String
+        let email: String
+        let first_name: String
+        let is_active: Int
+        let id: Int
+        let is_superuser: Int
+        let last_login: String
+        let last_name: String
+        let resource_uri: String
+        let username: String
+    }
+    
+    struct UserProfileStruct: Decodable {
```

Comment:
Don't we already have a UserProfile class in UserModel.swift? Can you use that instead? If that's not decodable, you can change that to a struct.

Also move all these struct to UserModel.swift file for cleaness.